.github/workflows/_tags.yml

```
@@ -0,0 +1,104 @@
+name: Generate and fetch tags
+on:
+  workflow_call:
+    inputs:
+      github_event_name:
+        type: string
+        required: true
+    outputs:
+      build-tag:
+        description: "Tag for the current workflow run"
+        value: ${{ jobs.tags.outputs.build-tag }}
+      previous-storage-release:
+        description: "Tag of the last storage release"
+        value: ${{ jobs.tags.outputs.storage }}
+      previous-proxy-release:
+        description: "Tag of the last proxy release"
+        value: ${{ jobs.tags.outputs.proxy }}
+      previous-compute-release:
+        description: "Tag of the last compute release"
+        value: ${{ jobs.tags.outputs.compute }}
+      run-kind:
+        description: "The kind of run we're currently in. Will be one of `pr-main`, `push-main`, `storage-rc`, `storage-release`, `proxy-rc`, `proxy-release`, `compute-rc`, `compute-release` or `merge_queue`"
+        value: ${{ jobs.tags.outputs.run-kind }}
+
+permissions: {}
+
+jobs:
+  tags:
+    runs-on: ubuntu-22.04
+    outputs:
+      build-tag: ${{ steps.build-tag.outputs.tag }}
+      compute: ${{ steps.previous-releases.outputs.compute }}
+      proxy: ${{ steps.previous-releases.outputs.proxy }}
+      storage: ${{ steps.previous-releases.outputs.storage }}
+      run-kind: ${{ steps.run-kind.outputs.run-kind }}
+    permissions:
+      contents: read
+    steps:
+      # Need `fetch-depth: 0` to count the number of commits in the branch
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get run kind
+        id: run-kind
+        env:
+          RUN_KIND: >-
+            ${{
+              false
+              || (inputs.github_event_name == 'push'         && github.ref_name == 'main')            && 'push-main'
+              || (inputs.github_event_name == 'push'         && github.ref_name == 'release')         && 'storage-release'
+              || (inputs.github_event_name == 'push'         && github.ref_name == 'release-compute') && 'compute-release'
+              || (inputs.github_event_name == 'push'         && github.ref_name == 'release-proxy')   && 'proxy-release'
+              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'main')            && 'pr-main'
+              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release')         && 'storage-rc'
+              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release-compute') && 'compute-rc'
+              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release-proxy')   && 'proxy-rc'
```

An optional one: I feel that having a `-pr` in the string makes it easier to understand that's it's about release PR

```suggestion
              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release')         && 'storage-rc-pr'
              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release-compute') && 'compute-rc-pr'
              || (inputs.github_event_name == 'pull_request' && github.base_ref == 'release-proxy')   && 'proxy-rc-pr'
```